src/Resources/contao/modules/ModuleNewsArchive.php

```diff
@@ -186,14 +186,39 @@ protected function compile()
 			}
 		}
 
+		// Determine sorting
+		$t = \NewsModel::getTable();
+		$arrOptions = array();
+		switch ($this->news_sorting)
+		{
+			case 'sort_date_asc':
+				$arrOptions['order'] = "$t.date ASC";
+				break;
+
+			case 'sort_headline_asc':
+				$arrOptions['order'] = "$t.headline ASC";
+				break;
+
+			case 'sort_headline_desc':
+				$arrOptions['order'] = "$t.headline DESC";
+				break;
+
+			case 'sort_random':
```

Comment:
Yes, we needed it a couple of times at least. Currently in two projects as well.